autoprecompiles/src/constraint_optimizer.rs

```diff
@@ -185,6 +188,17 @@ fn remove_trivial_constraints<P: FieldElement>(
     symbolic_machine
 }
 
+fn remove_equal_constraints<P: FieldElement>(
```

Comment:
Maybe we could extract this into a separate PR? Seems unrelated, and would be interesting to see the effect of just this.

Also, we could do the same for stateless bus interactions!

Reply:
Yes, it's from here: https://github.com/powdr-labs/powdr/pull/2777